### PR TITLE
chore: update torch v2.5.1

### PR DESCRIPTION
**Pull request by zhyncs**
## Motivation

ref https://github.com/sgl-project/sglang/actions/runs/11605237798/job/32315285934

## Checklist

- [ ] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by merrymercy** [comment]
cc @jerryzh168  who needs torch 2.5

File: python/pyproject.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ runtime_common = ["aiohttp", "decord", "fastapi", "hf_transfer", "huggingface_hu
     "outlines>=0.0.44", "modelscope"]
 # xpu is not enabled in public vllm and torch whl,
 # need to follow https://docs.vllm.ai/en/latest/getting_started/xpu-installation.htmlinstall vllm
-srt = ["sglang[runtime_common]", "torch", "vllm==0.6.3.post1"]
+srt = ["sglang[runtime_common]", "torch>=2.5.1", "vllm==0.6.3.post1; python_full_version>'0.0.0'"]
```
Comment (jerryzh168) [inline python/pyproject.toml]: does vllm 0.6.3.post1 mean the most recent branch? they just updated to 2.5 a few days ago: https://github.com/vllm-project/vllm/pull/9588

Comment (zhyncs) [reply python/pyproject.toml]: I need to wait vLLM to release the new version

**Comment by fengyang95** [comment]
@zhyncs In this case, does torch compile now support fp8?

**Comment by zhyncs** [comment]
https://github.com/sgl-project/sglang/actions/runs/11853855192

**Comment by zhyncs** [comment]
https://github.com/sgl-project/sglang/actions/runs/11870636139

**Comment by zhyncs** [comment]
https://github.com/sgl-project/sglang/actions/runs/11872674590

**Comment by zhyncs** [comment]
https://github.com/sgl-project/sglang/actions/runs/11878956634

File: test/srt/test_nightly_gsm8k_eval.py [hunk 2]
```diff
@@ -14,12 +17,35 @@
     popen_launch_server,
 )
 
+# Define score thresholds for each model (99% of the previous score)
```
Comment (zhyncs) [inline test/srt/test_nightly_gsm8k_eval.py]: The previous default threshold of 0.5 was too low, but due to the instability of eval, setting a single value can easily lead to failure. Therefore, we choose to set it at 99% here, which is stricter than the original 0.5.

File: python/sglang/srt/layers/activation.py [hunk 3]
```diff
@@ -38,6 +38,7 @@
 logger = logging.getLogger(__name__)
 
 
+@CustomOp.register("silu_and_mul")
```
Comment (zhyncs) [inline python/sglang/srt/layers/activation.py:41]: To be compatible with the new version

File: python/sglang/srt/model_executor/model_runner.py [hunk 4]
```diff
@@ -243,12 +245,14 @@ def load_model(self):
...
         # Prepare the vllm model config
         monkey_patch_vllm_dummy_weight_loader()
+        monkey_patch_vllm_model_config()
         self.load_config = LoadConfig(
             load_format=self.server_args.load_format,
             download_dir=self.server_args.download_dir,
         )
         self.vllm_model_config = VllmModelConfig(
             model=self.server_args.model_path,
+            task="generate" if self.model_config.is_generation else "embedding",
```
Comment (zhyncs) [inline python/sglang/srt/model_executor/model_runner.py:255]: `task` field is required

File: python/sglang/srt/model_executor/model_runner.py [hunk 5]
```diff
@@ -263,15 +267,17 @@ def load_model(self):
             )
         self.dtype = self.vllm_model_config.dtype
 
+        self.vllm_config = VllmConfig()
+        self.vllm_config.model_config = self.vllm_model_config
+        self.vllm_config.load_config = self.load_config
+        self.vllm_config.device_config = DeviceConfig(self.device)
+        self.vllm_config.quant_config = VllmConfig._get_quantization_config(
```
Comment (zhyncs) [inline python/sglang/srt/model_executor/model_runner.py:274]: We need to explicitly set the quant config.

File: python/sglang/srt/utils.py [hunk 6]
```diff
@@ -512,6 +498,60 @@ def maybe_set_triton_cache_manager() -> None:
...
+            "MistralModel",
+            "LlamaForSequenceClassification",
+            "LlamaForSequenceClassificationWithNormal_Weights",
+            "InternLM2ForRewardModel",
+        }
+
+        is_generation = not any(arch in non_generation_models for arch in architectures)
+
+        auto_map = getattr(hf_config, "auto_map", {})
+        has_sequence_classification = any(
```
Comment (zhyncs) [inline python/sglang/srt/utils.py:528]: Here is a workaround for the Qwen embedding model

File: test/srt/test_nightly_gsm8k_eval.py [hunk 7]
```diff
@@ -14,12 +17,35 @@
...
+MODEL_SCORE_THRESHOLDS = {
```
Comment (zhyncs) [inline test/srt/test_nightly_gsm8k_eval.py:20]: This list includes the currently most popular open-source models such as Llama, Mistral, Mixtral, DeepSeek, Qwen, and also commonly used quant models like AWQ, GPTQ, compressed-tensor. Subsequent updates may include corresponding versions like Llama 3.1 -> Llama 3.2 or Qwen 2 -> Qwen 2.5 and so on.

**Comment by zhyncs** [comment]
Currently, the performance of non-streaming small batch size cases is below expectations. ref https://github.com/sgl-project/sglang/pull/1563 cc @Ying1123

File: test/srt/test_nightly_gsm8k_eval.py (hunk 7 above)
Comment (zhyncs) [reply test/srt/test_nightly_gsm8k_eval.py:20]: Using 0.99 is sometimes still unstable, I plan to replace the current gsm8k eval with `lm_eval` in the follow-up PR.

**Comment by zhyncs** [comment]
https://github.com/sgl-project/sglang/actions/runs/11879840842

**Comment by zhyncs** [comment]
```
[
  {
    "timestamp": "2024-11-17T07:25:49.048134",
    "model": "meta-llama/Llama-3.1-8B-Instruct",
    "metrics": {
      "en": 0.836,
      "en:std": 0.3702755730533679,
      "group_latin": 0.836,
      "group_latin:std": 0.3702755730533679,
      "score:std": 0.3702755730533679,
      "score": 0.836
    },
    "score": 0.836
  },
  {
    "timestamp": "2024-11-17T07:26:37.000698",
    "model": "mistralai/Mistral-7B-Instruct-v0.3",
    "metrics": {
      "en": 0.604,
      "en:std": 0.48906441293555597,
      "group_latin": 0.604,
      "group_latin:std": 0.48906441293555597,
      "score:std": 0.48906441293555597,
      "score": 0.604
    },
    "score": 0.604
  },
  {
    "timestamp": "2024-11-17T07:28:02.777706",
    "model": "deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct",
    "metrics": {
      "en": 0.876,
      "en:std": 0.3295815528818322,
      "group_latin": 0.876,
      "group_latin:std": 0.3295815528818322,
      "score:std": 0.3295815528818322,
      "score": 0.876
    },
    "score": 0.876
  },
  {
    "timestamp": "2024-11-17T07:29:14.852606",
    "model": "google/gemma-2-27b-it",
    "metrics": {
      "en": 0.924,
      "en:std": 0.26499811320083017,
      "group_latin": 0.924,
      "group_latin:std": 0.26499811320083017,
      "score:std": 0.26499811320083017,
      "score": 0.924
    },
    "score": 0.924
  },
  {
    "timestamp": "2024-11-17T07:32:26.439859",
    "model": "meta-llama/Llama-3.1-70B-Instruct",
    "metrics": {
      "en": 0.976,
      "en:std": 0.15304901175767194,
      "group_latin": 0.976,
      "group_latin:std": 0.15304901175767194,
      "score:std": 0.15304901175767194,
      "score": 0.976
    },
    "score": 0.976
  },
  {
    "timestamp": "2024-11-17T07:34:16.554962",
    "model": "mistralai/Mixtral-8x7B-Instruct-v0.1",
    "metrics": {
      "en": 0.648,
      "en:std": 0.4775939698111775,
      "group_latin": 0.648,
      "group_latin:std": 0.4775939698111775,
      "score:std": 0.4775939698111775,
      "score": 0.648
    },
    "score": 0.648
  },
  {
    "timestamp": "2024-11-17T07:35:54.798802",
    "model": "Qwen/Qwen2-57B-A14B-Instruct",
    "metrics": {
      "en": 0.884,
      "en:std": 0.32022492095400695,
      "group_latin": 0.884,
      "group_latin:std": 0.32022492095400695,
      "score:std": 0.32022492095400695,
      "score": 0.884
    },
    "score": 0.884
  },
  {
    "timestamp": "2024-11-17T07:37:30.045613",
    "model": "deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct",
    "metrics": {
      "en": 0.86,
      "en:std": 0.3469870314579494,
      "group_latin": 0.86,
      "group_latin:std": 0.3469870314579494,
      "score:std": 0.3469870314579494,
      "score": 0.86
    },
    "score": 0.86
  },
  {
    "timestamp": "2024-11-17T07:38:26.084096",
    "model": "neuralmagic/Meta-Llama-3.1-8B-Instruct-FP8",
    "metrics": {
      "en": 0.88,
      "en:std": 0.32496153618543844,
      "group_latin": 0.88,
      "group_latin:std": 0.32496153618543844,
      "score:std": 0.32496153618543844,
      "score": 0.88
    },
    "score": 0.88
  },
  {
    "timestamp": "2024-11-17T07:39:14.548670",
    "model": "neuralmagic/Mistral-7B-Instruct-v0.3-FP8",
    "metrics": {
      "en": 0.552,
      "en:std": 0.4972886485734417,
      "group_latin": 0.552,
      "group_latin:std": 0.4972886485734417,
      "score:std": 0.4972886485734417,
      "score": 0.552
    },
    "score": 0.552
  },
  {
    "timestamp": "2024-11-17T07:40:51.686684",
    "model": "neuralmagic/DeepSeek-Coder-V2-Lite-Instruct-FP8",
    "metrics": {
      "en": 0.888,
      "en:std": 0.31536645351083237,
      "group_latin": 0.888,
      "group_latin:std": 0.31536645351083237,
      "score:std": 0.31536645351083237,
      "score": 0.888
    },
    "score": 0.888
  },
  {
    "timestamp": "2024-11-17T07:41:42.733429",
    "model": "neuralmagic/gemma-2-2b-it-FP8",
    "metrics": {
      "en": 0.612,
      "en:std": 0.4872945721019269,
      "group_latin": 0.612,
      "group_latin:std": 0.4872945721019269,
      "score:std": 0.4872945721019269,
      "score": 0.612
    },
    "score": 0.612
  },
  {
    "timestamp": "2024-11-17T07:43:15.627071",
    "model": "neuralmagic/Meta-Llama-3.1-70B-Instruct-FP8",
    "metrics": {
      "en": 0.964,
      "en:std": 0.18629009635512025,
      "group_latin": 0.964,
      "group_latin:std": 0.18629009635512025,
      "score:std": 0.18629009635512025,
      "score": 0.964
    },
    "score": 0.964
  },
  {
    "timestamp": "2024-11-17T07:44:52.078680",
    "model": "neuralmagic/Mixtral-8x7B-Instruct-v0.1-FP8",
    "metrics": {
      "en": 0.624,
      "en:std": 0.48438001610305936,
      "group_latin": 0.624,
      "group_latin:std": 0.48438001610305936,
      "score:std": 0.48438001610305936,
      "score": 0.624
    },
    "score": 0.624
  },
  {
    "timestamp": "2024-11-17T07:46:25.651174",
    "model": "neuralmagic/Qwen2-72B-Instruct-FP8",
    "metrics": {
      "en": 0.948,
      "en:std": 0.2220270253820467,
      "group_latin": 0.948,
      "group_latin:std": 0.2220270253820467,
      "score:std": 0.2220270253820467,
      "score": 0.948
    },
    "score": 0.948
  },
  {
    "timestamp": "2024-11-17T07:48:11.049916",
    "model": "neuralmagic/Qwen2-57B-A14B-Instruct-FP8",
    "metrics": {
      "en": 0.824,
      "en:std": 0.380820167533181,
      "group_latin": 0.824,
      "group_latin:std": 0.380820167533181,
      "score:std": 0.380820167533181,
      "score": 0.824
    },
    "score": 0.824
  },
  {
    "timestamp": "2024-11-17T07:50:01.784188",
    "model": "neuralmagic/DeepSeek-Coder-V2-Lite-Instruct-FP8",
    "metrics": {
      "en": 0.876,
      "en:std": 0.3295815528818322,
      "group_latin": 0.876,
      "group_latin:std": 0.3295815528818322,
      "score:std": 0.3295815528818322,
      "score": 0.876
    },
    "score": 0.876
  },
  {
    "timestamp": "2024-11-17T07:51:11.661882",
    "model": "hugging-quants/Meta-Llama-3.1-8B-Instruct-AWQ-INT4",
    "metrics": {
      "en": 0.848,
      "en:std": 0.35902089075707005,
      "group_latin": 0.848,
      "group_latin:std": 0.35902089075707005,
      "score:std": 0.35902089075707005,
      "score": 0.848
    },
    "score": 0.848
  },
  {
    "timestamp": "2024-11-17T07:52:26.298293",
    "model": "hugging-quants/Meta-Llama-3.1-8B-Instruct-GPTQ-INT4",
    "metrics": {
      "en": 0.852,
      "en:std": 0.35509998591945907,
      "group_latin": 0.852,
      "group_latin:std": 0.35509998591945907,
      "score:std": 0.35509998591945907,
      "score": 0.852
    },
    "score": 0.852
  }
]
```
The results of the nightly eval gsm8k on NVIDIA Cloud H100 are OK, but the CIs are not very stable, so ignore them for now.

**Comment by zhyncs** [comment]
BTW I only verified on NVIDIA Cloud H100, may you help verify on AMD and Intel  cc @HaiShaw @liangan1

**Comment by zhyncs** [comment]
After this PR is merged, all other CIs passed except https://github.com/sgl-project/sglang/actions/runs/11880304446/job/33103293929
```
Traceback (most recent call last):
  File "/actions-runner/_work/sglang/sglang/test/srt/test_triton_attention_backend.py", line 29, in test_latency
Writing report to /tmp/mmlu_meta-llama_Llama-3.1-8B-Instruct.html
    assert output_throughput > 153, f"{output_throughput=}"
AssertionError: output_throughput=151.8
```
The results are very close and can be temporarily ignored, use another issue https://github.com/sgl-project/sglang/issues/2059 to track.

**Pull request merged**